framework/web/Request.php

```diff
@@ -842,8 +950,8 @@ public function setUrl($value)
      */
     protected function resolveRequestUri()
     {
-        if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { // IIS
-            $requestUri = $_SERVER['HTTP_X_REWRITE_URL'];
+        if ($this->headers->has('X-Rewrite-Url')) { // IIS
```

Comment:
good point, I have no idea how IIS works tough, this would be another BC break for systems that use this header.